public/app/features/explore/utils/ResultProcessor.ts

```diff
@@ -50,7 +50,20 @@ export class ResultProcessor {
       return null;
     }
 
-    const onlyTables = this.dataFrames.filter(frame => shouldShowInVisualisationType(frame, 'table'));
+    const onlyTables = this.dataFrames
+      .filter((frame: DataFrame) => shouldShowInVisualisationType(frame, 'table'))
+      .sort((frameA: DataFrame, frameB: DataFrame) => {
+        const frameARefId = frameA.refId!;
+        const frameBRefId = frameB.refId!;
+
+        if (frameARefId > frameBRefId) {
```

Comment:
How do we create the refs? As far as I am aware they are just string right? Are they always created in a way that this sort hold, ie don't we create them at random?

Reply:
Yes, they are strings. They come from the query field and are in order (alphabetical order according to the order of queries in Explore - as same as with queries in dashboards).

Reply:
Yes but in this function https://github.com/grafana/grafana/blob/ae09ccbf7977c0c5e5b9f8a2b431582816eca82a/public/app/core/utils/query.ts#L4 it's not starting from the latest value it always start from A. So if you add two queries you have AB, then delete the first one you have B, add a new row you have BA and this sort will sort in the other way around from what is on the screen. Wouldn't it make sense to sort it based on what the ordering is in the reducer and not count on the refIDs being sequential?